ast/compile.go

```diff
@@ -1573,6 +1579,46 @@ func (vis *bodySafetyVisitor) checkArrayComprehensionSafety(ac *ArrayComprehensi
 	}
 }
 
+func (vis *bodySafetyVisitor) checkObjectComprehensionSafety(oc *ObjectComprehension) {
```

Comment:
These three checks share a lot of code. Did you think about refactoring to avoid duplication?